R/boundary_designs.R

```diff
@@ -0,0 +1,145 @@
+#' Boundary designs
+#'
+#' Compute boundary designs. These can be used for optimization and are
+#' implemented by default in \link{minimize}.
+#'
+#' @param objective An object of class \code{\link{UnconditionalScore-class}} to be minimized
+#' @param subject_to A \code{\link{ConstraintsCollection-class}} defining the constraints
+#' @param initial_design The initial design
+#' @param ... optimal arguments
+#'
+#' @rdname boundary-designs
+#' @export
+setGeneric("get_lower_boundary_design",
+           function(objective, subject_to, initial_design, ...) standardGeneric("get_lower_boundary_design"))
+
+
+
+#' @rdname boundary-designs
+#' @export
+setGeneric("get_upper_boundary_design",
+           function(objective, subject_to, initial_design, ...) standardGeneric("get_upper_boundary_design"))
+
+
+
+#' @rdname boundary-designs
+#' @export
+setMethod("get_lower_boundary_design",
+          signature("UnconditionalScore", "ConstraintsCollection", "OneStageDesign"),
+          function(objective, subject_to, initial_design, ...) {
+              OneStageDesign(5, 0.0)
+          })
+
+
+#' @rdname boundary-designs
+#' @export
+setMethod("get_lower_boundary_design",
+          signature("UnconditionalScore", "ConstraintsCollection", "GroupSequentialDesign"),
+          function(objective, subject_to, initial_design, ...) {
+              cnstrs <- subject_to
+              cnstrs@conditional_constraints <- list()
+
+              os_design <- minimize(
```

Comment:
v.s. i don't see the need to do optimization for the lower/upper boundaries, maybe if you want to suggest an initial design, but not for the bounds, what do you gain?